src/Eventso.Subscription.Kafka/Event.cs

```diff
@@ -59,5 +59,21 @@ public Event(ConsumeResult<Guid, ConsumedMessage> @event, string topic)
 
             return new[] { offset, resultPair };
         }
+
+        internal TopicPartitionOffset GetTopicPartitionOffset()
+            => new(Topic, Partition, Offset);
+
+        public bool Equals(Event other)
+            => _value.Equals(other._value)
+               && _key.Equals(other._key)
+               && Topic == other.Topic
+               && Partition.Equals(other.Partition)
+               && Offset.Equals(other.Offset);
```

Comment:
this order of comparison is not the best. Should be: offset key partition topic value

Reply:
no more IEquatable here